creator-node/src/routes/nodeSync.js

```diff
@@ -67,6 +68,25 @@ module.exports = function (app) {
       const ipfs = req.app.get('ipfsAPI')
       let ipfsIDObj = await getIPFSPeerId(ipfs, config)
 
+      // Ensure all relevant files are available through IPFS at export time
+      await Promise.all(files.map(async (file) => {
+        if (file.type === 'track' || file.type === 'metadata' || file.type === 'copy320') {
+          await rehydrateIpfsFromFsIfNecessary(
+            req,
+            file.multihash,
+            file.storagePath)
+        } else if (file.type === 'image') {
+          let sourcePath = file.sourceFile
+          if (sourcePath.includes('blob')) {
+            sourcePath = file.sourceFile.split('/')[1]
+          }
+          await rehydrateIpfsFromFsIfNecessary(
+            req,
+            file.multihash,
+            file.storagePath,
+            sourcePath)
```

Comment:
you call rehydrate every time you see an image file, but seems every time you call rehydrate with "dir" mode it will fetch all images in dir and re-add so seems like it would do multiple duplicate calls no?

also - is this back-compat? for images that were added before multi-res rehydrate would still treat them as being in a dir right?

Reply:
yes it would dup call and not sure about the back-compat, lets discuss it

Reply:
fixed